<html>
<head><meta charset="utf-8"><title>âœ” pyDataverse missing attribute in NativeAPI Â· python Â· Zulip Chat Archive</title></head>
<h2>Stream: <a href="https://gdcc.github.io/zulip-archive/stream/377090-python/index.html">python</a></h2>
<h3>Topic: <a href="https://gdcc.github.io/zulip-archive/stream/377090-python/topic/.E2.9C.94.20pyDataverse.20missing.20attribute.20in.20NativeAPI.html">âœ” pyDataverse missing attribute in NativeAPI</a></h3>

<hr>

<base href="https://dataverse.zulipchat.com">

<head><link href="https://gdcc.github.io/zulip-archive/style.css" rel="stylesheet"></head>

<a name="477481060"></a>
<h4><a href="https://dataverse.zulipchat.com#narrow/stream/377090-python/topic/%E2%9C%94%20pyDataverse%20missing%20attribute%20in%20NativeAPI/near/477481060" class="zl"><img src="https://gdcc.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Sherry Lake <a href="https://gdcc.github.io/zulip-archive/stream/377090-python/topic/.E2.9C.94.20pyDataverse.20missing.20attribute.20in.20NativeAPI.html#477481060">(Oct 17 2024 at 16:22)</a>:</h4>
<p>I am trying to execute:<br>
<code> api.dataverse_tree_walker(tree, dv_keys = ['dataverse_id','dataverse_alias'], ds_keys = ['dataset_id','pid'])</code></p>
<p>But am getting the error:</p>
<blockquote>
<p>AttributeError: 'NativeApi' object has no attribute 'dataverse_tree_walker'</p>
</blockquote>
<p>I have upgraded and installed V 0.3.4 pyDataverse</p>
<blockquote>
<p>Name: pyDataverse<br>
Version: 0.3.4</p>
</blockquote>
<p>Have imported NativeAPI: </p>
<p><code>dir()</code></p>
<blockquote>
<p>['DataAccessApi', 'Dataset', 'NativeApi', '__annotations__', '__builtins__', '__doc__', '__loader__', '__name__', '__package__', '__spec__', 'api', 'data_api', 'read_file', 'resp', 'tree']</p>
</blockquote>
<p>I have confirmed "tree_walker" is not in "NativeApi". Am I missing a import?</p>



<a name="477511499"></a>
<h4><a href="https://dataverse.zulipchat.com#narrow/stream/377090-python/topic/%E2%9C%94%20pyDataverse%20missing%20attribute%20in%20NativeAPI/near/477511499" class="zl"><img src="https://gdcc.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Philip Durbin ðŸš€ <a href="https://gdcc.github.io/zulip-archive/stream/377090-python/topic/.E2.9C.94.20pyDataverse.20missing.20attribute.20in.20NativeAPI.html#477511499">(Oct 17 2024 at 19:33)</a>:</h4>
<p>In <a href="https://github.com/gdcc/pyDataverse/blob/v0.3.4/tests/utils/test_utils.py">a test</a> I see <code>from pyDataverse.utils import read_json, dataverse_tree_walker</code> so I guess it's in <code>utils</code>.</p>



<a name="477515796"></a>
<h4><a href="https://dataverse.zulipchat.com#narrow/stream/377090-python/topic/%E2%9C%94%20pyDataverse%20missing%20attribute%20in%20NativeAPI/near/477515796" class="zl"><img src="https://gdcc.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Sherry Lake <a href="https://gdcc.github.io/zulip-archive/stream/377090-python/topic/.E2.9C.94.20pyDataverse.20missing.20attribute.20in.20NativeAPI.html#477515796">(Oct 17 2024 at 20:03)</a>:</h4>
<p>Thanks. I saw something about "utils", but I was trying to import "utils" from pyDataverse.api adding "utils" to this line - didn't work.</p>
<p><code>from pyDataverse.api import NativeApi, DataAccessApi, utils</code></p>
<p>I don't think the pyDataverse.utils is explained. But now I know what to look for. It's working for me.</p>
<p><img alt=":dataverse_man:" class="emoji" src="https://avatars.zulip.com/53090/emoji/images/c7b94a03.png" title="dataverse man"></p>



<a name="477515834"></a>
<h4><a href="https://dataverse.zulipchat.com#narrow/stream/377090-python/topic/%E2%9C%94%20pyDataverse%20missing%20attribute%20in%20NativeAPI/near/477515834" class="zl"><img src="https://gdcc.github.io/zulip-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Notification Bot <a href="https://gdcc.github.io/zulip-archive/stream/377090-python/topic/.E2.9C.94.20pyDataverse.20missing.20attribute.20in.20NativeAPI.html#477515834">(Oct 17 2024 at 20:04)</a>:</h4>
<p><span class="user-mention silent" data-user-id="599826">Sherry Lake</span> has marked this topic as resolved.</p>



<hr><p>Last updated: Nov 01 2025 at 23:12 UTC</p>
</html>